metaflow/plugins/kfp/kfp.py

```diff
@@ -404,27 +431,27 @@ def step_op_func(
     print("----")
 
     # TODO: Map username to KFP specific user/profile/namespace
+    # Note: we don't put this in a try catch block as below because
+    # Popen will not produce a error that will cause the Kubeflow step
+    # to stop
     with Popen(
-        ["/bin/sh", "-c", cmd],
-        stdin=PIPE,
-        stdout=PIPE,
-        stderr=STDOUT,
+        cmd,
+        shell=True,
         universal_newlines=True,
+        executable="/bin/bash",
         env=dict(os.environ, USERNAME="kfp-user", METAFLOW_RUN_ID=kfp_run_id),
-    ) as process, StringIO() as string_buffer:
-        for line in process.stdout:
-            print(line, end="")
-            string_buffer.write(line)
-        buffer_output = string_buffer.getvalue()
-    print("___ DONE ___")
-
-    if process.returncode != 0:
-        raise Exception("Returned: %s" % process.returncode)
-
-    with open(
-        os.path.join(tempfile.gettempdir(), "kfp_metaflow_out_dict.json"), "r"
-    ) as file:
-        task_out_dict = json.load(file)
+    ) as process:
+        print("Running command.")
+    process.wait()
+
+    try:
```

Comment:
Since I chained the actual step_cli commands with the `aws cp` commands with a `;` instead of `&`, `process.returncode` will always return 0. This is because `;` causes subsequent commands even if the current command fails to run and always returns a 0 return code. Thus the conditional statement `if process.returncode != 0` won't work. I've tried looking into alternative solutions, namely `set -o pipefail` (https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html#The-Set-Builtin), but this only returns a nonzero exit code if the commands are combined with '&&'.

Reply:
Why must the step_cli commands be chained with `aws cp` with a `:`?  Why can't you combine with `&&` or `&`?

Reply:
The reason is that if the step_cli commands fail, then the `aws cp` won't execute unless we use `;`. `;` allows a subsequent command run even if the previous command fails. See the following cheatsheet:
```
A; B    # Run A and then B, regardless of success of A
A && B  # Run B if and only if A succeeded
A || B  # Run B if and only if A failed
A &     # Run A in background.
```